src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/RequesterNameAndAddress.java

```diff
@@ -20,31 +29,40 @@
     private static final String KEY = "NAD";
     private static final String QUALIFIER = "PO";
     public static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
-    private static final int REQUESTER_NAME_INDEX_IN_EDIFACT_STRING = 4;
 
-    @NonNull
     private final String identifier;
-    @NonNull
-    private final HealthcareRegistrationIdentificationCode healthcareRegistrationIdentificationCode;
-
-    private final String requesterName;
+    private final HealthcareRegistrationIdentificationCode code;
+    private final String practitionerName;
+    private final String organizationName;
 
+    @SuppressWarnings("checkstyle:MagicNumber")
     public static RequesterNameAndAddress fromString(final String edifactString) {
         if (!edifactString.startsWith(KEY_QUALIFIER)) {
             throw new IllegalArgumentException("Can't create " + RequesterNameAndAddress.class.getSimpleName()
                 + " from " + edifactString);
         }
 
-        final String[] keySplit = Split.byPlus(edifactString);
-        final String identifier = Split.byColon(keySplit[2])[0];
-        final String code = Split.byColon(keySplit[2])[1];
-        final String requesterName = keySplit[REQUESTER_NAME_INDEX_IN_EDIFACT_STRING];
+        String[] keySplit = Split.byPlus(edifactString);
+        String identifier = Split.byColon(keySplit[2])[0];
+
+        final boolean isOrganization = StringUtils.isBlank(identifier);
+        if (isOrganization) {
+            // if identifier is blank - organization
+            String organizationName = keySplit[4].replaceAll("\\?'", "'");
```

Comment:
We discussed this before and agreed that the segments should hold the raw data and the mappers would be responsible for converting and escaping them.